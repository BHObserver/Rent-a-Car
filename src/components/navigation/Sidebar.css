.sidebar {
    display: flex;
    flex-direction: column;
    background-color: whitesmoke;
    width: 200px;
    height: 100%;
    position: fixed;
    top: 0px;
    left: 0px;
    overflow: auto;
    width: 12%;
    border-right: 1px solid gray;
  }
  
  .navigation {
    display: flex;
    flex-direction: column;
    justify-content: center;
    align-items: center;
    width: 215px;
    position: fixed;
    top: 35%;
    left: 0;
  }
  
  .sidebar a {
    text-decoration: none;
    color: black;
    display: block;
  }
  
  .navigation > ul {
    list-style-type: none;
    margin: 0;
    padding: 0;
  }
  
  .navigation > ul > li > a {
    padding: 20px; 
  }
  
  .navigation > ul > li > a:hover {
    background-color: #6a9e09;
    color: #f8f9f9;
  }
  
  .navigation li:last-child {
    border: none;
  }
  
  .contents {
    margin-left: 12%;
    padding: 5px;
  }
  
  .logo {
    font-size: 24px;
    text-decoration: none;
    padding: 20px 50px 0; 
    display: flex;
    flex-direction: column;
    justify-content: center;
    align-items: center;
    width: 215px;
    position: fixed;
    top: 5%;
    left: 0;
  }
  