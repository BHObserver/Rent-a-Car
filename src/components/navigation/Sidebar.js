import React, { useState } from 'react';
import { NavLink } from 'react-router-dom';
import {
  Drawer, List, ListItem, ListItemIcon, ListItemText, Typography, Divider, IconButton,
} from '@mui/material';
import { styled } from '@mui/system';
import MenuIcon from '@mui/icons-material/Menu';
import HomeIcon from '@mui/icons-material/Home';
import EventIcon from '@mui/icons-material/Event';
import DirectionsCarIcon from '@mui/icons-material/DirectionsCar';
import PlaylistAddIcon from '@mui/icons-material/PlaylistAdd';
import DeleteIcon from '@mui/icons-material/Delete';
import LogoutIcon from '@mui/icons-material/Logout';
import FacebookIcon from '@mui/icons-material/Facebook';
import TwitterIcon from '@mui/icons-material/Twitter';
import InstagramIcon from '@mui/icons-material/Instagram';
import { useDispatch } from 'react-redux';
import { logout } from '../../redux/session/actions/authActions';
import logo from '../../assets/images/logo.png';

const StyledList = styled(List)({
  display: 'flex',
});

const Sidebar = () => {
  const dispatch = useDispatch();
  const [open, setOpen] = useState(false);

  const handleLogout = () => {
    dispatch(logout());
  };

  return (
    <>
      <IconButton
        color="inherit"
        aria-label="open drawer"
        onClick={() => setOpen(!open)}
        sx={{
          display: 'flex',
          zIndex: 1,
          alignItems: 'flex-start', // Position sidebar to the top-left corner
          justifyContent: 'flex-start',
          width: '20px',
          height: '20px',
          padding: '10px',
          color: '#fff',
        }} // Adjust position as needed
      >
        <MenuIcon />
      </IconButton>
      <Drawer
        variant="temporary"
        open={open}
        onClose={() => setOpen(false)}
        sx={{
          width: 250,
          flexShrink: 0,
          '& .MuiDrawer-paper': {
            width: 250,
            boxSizing: 'border-box',
            background: 'linear-gradient(180deg, #009688 0%, #00796B 100%)', // Gradient background
            color: '#fff', // Text color
          },
        }}
      >
        <Typography variant="h6" align="center" sx={{ pt: 2, pb: 2 }}>
          <div className="logo-container">
            <img src={logo} alt="Logo" style={{ width: '100px', height: 'auto' }} />
            <h2>RENT A CAR</h2>
          </div>
          {' '}
        </Typography>
        <Divider />
        <List>
          <ListItem button component={NavLink} to="/profile" activeClassName="active">
            <ListItemIcon><HomeIcon /></ListItemIcon>
            <ListItemText primary="Home" />
          </ListItem>
          <ListItem button component={NavLink} to="/reserve" activeClassName="active">
            <ListItemIcon><DirectionsCarIcon /></ListItemIcon>
            <ListItemText primary="Reserve" />
          </ListItem>
          <ListItem button component={NavLink} to="/my-reservations" activeClassName="active">
            <ListItemIcon><EventIcon /></ListItemIcon>
            <ListItemText primary="My Reservations" />
          </ListItem>
          <ListItem button component={NavLink} to="/add-item" activeClassName="active">
            <ListItemIcon><PlaylistAddIcon /></ListItemIcon>
            <ListItemText primary="Add Car" />
          </ListItem>
          <ListItem button component={NavLink} to="/delete-item" activeClassName="active">
            <ListItemIcon><DeleteIcon /></ListItemIcon>
            <ListItemText primary="Delete Car" />
          </ListItem>
        </List>
        <Divider />
        <List>
          <ListItem button component={NavLink} onClick={handleLogout} to="/login">
            <ListItemIcon><LogoutIcon /></ListItemIcon>
            <ListItemText primary="Logout" />
          </ListItem>
        </List>
        {/* Social Media Icons */}
        <StyledList>
          <ListItem button component="a" href="https://www.facebook.com">
            <ListItemIcon><FacebookIcon /></ListItemIcon>
          </ListItem>
          <ListItem button component="a" href="https://www.twitter.com">
            <ListItemIcon><TwitterIcon /></ListItemIcon>
          </ListItem>
          <ListItem button component="a" href="https://www.instagram.com">
            <ListItemIcon><InstagramIcon /></ListItemIcon>
          </ListItem>
        </StyledList>
      </Drawer>
    </>
  );
};

export default Sidebar;
