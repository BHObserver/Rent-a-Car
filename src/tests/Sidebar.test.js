import React from 'react';
import { BrowserRouter as Router } from 'react-router-dom';
import renderer from 'react-test-renderer';
import { useDispatch } from 'react-redux';
import Sidebar from '../components/navigation/Sidebar';
// Mocking useDispatch hook
jest.mock('react-redux', () => ({
  ...jest.requireActual('react-redux'),
  useDispatch: jest.fn(),
}));

describe('Sidebar component', () => {
  it('renders correctly', () => {
    // Mocking useDispatch hook
    const dispatch = jest.fn();
    useDispatch.mockReturnValue(dispatch);

    const component = renderer.create(
      <Router>
        <Sidebar />
      </Router>,
    );
    const tree = component.toJSON();

    expect(tree).toMatchSnapshot();
  });
});
