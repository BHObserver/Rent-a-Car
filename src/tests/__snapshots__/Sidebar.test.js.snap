// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Sidebar component renders correctly 1`] = `
<button
  aria-label="open drawer"
  className="MuiButtonBase-root MuiIconButton-root MuiIconButton-colorInherit MuiIconButton-sizeMedium css-9v4n2c-MuiButtonBase-root-MuiIconButton-root"
  disabled={false}
  onBlur={[Function]}
  onClick={[Function]}
  onContextMenu={[Function]}
  onDragLeave={[Function]}
  onFocus={[Function]}
  onKeyDown={[Function]}
  onKeyUp={[Function]}
  onMouseDown={[Function]}
  onMouseLeave={[Function]}
  onMouseUp={[Function]}
  onTouchEnd={[Function]}
  onTouchMove={[Function]}
  onTouchStart={[Function]}
  tabIndex={0}
  type="button"
>
  <svg
    aria-hidden={true}
    className="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium css-i4bv87-MuiSvgIcon-root"
    data-testid="MenuIcon"
    focusable="false"
    viewBox="0 0 24 24"
  >
    <path
      d="M3 18h18v-2H3zm0-5h18v-2H3zm0-7v2h18V6z"
    />
  </svg>
</button>
`;
